src/coreComponents/codingUtilities/TableData.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file TableData.cpp
+ */
+
+#include "codingUtilities/TableData.hpp"
+
+namespace geos
+{
+
+void TableData::addRow( std::vector< string > const & row )
+{
+  m_rows.push_back( row );
+}
+
+void TableData::clear()
+{
+  m_rows.clear();
+}
+
+std::vector< std::vector< string > > const & TableData::getTableDataRows() const
+{
+  return m_rows;
+}
+
+std::set< real64 > const & TableData2D::getColumns() const
+{
+  return m_columns;
+}
+std::set< real64 > const & TableData2D::getRows() const
+{
+  return m_rows;
+}
+
+TableData TableData2D::buildTableData() const
+{
+  TableData tableDataToBeBuilt;
+  for( real64 const & rowValue : m_rows )
+  {
+    std::vector< string > values;
+    values.push_back( GEOS_FMT( "{}", rowValue ) );
+    for( real64 const & columnValue : m_columns )
+    {
+      std::pair< real64, real64 > id = std::pair< real64, real64 >( rowValue, columnValue );
```

Comment:
```suggestion
      std::pair< real64, real64 > const id = std::make_pair( rowValue, columnValue );
```
I'm not sure to understand.
Aren't you registering data in your map but the same data is stored in `std::set` alongside?
Then you loop over all the combinations to find which combinations are valid?
Couldn't you more simply loop on the map and order the entries?

Did you check what happens if you have duplicates in your entries?
The duplication will be lost since you're storing the entries in a `std::set`.

Reply:
This function `buildTableData` is used for building 2D table, there is an example in PVT Tables PR.

Reply:
The `buildTableData` function is used for building 2D table data. You can find an example of these :  [PVT Tables logging improvements](https://github.com/GEOS-DEV/GEOS/pull/3003/files#top) in TableFunction.cpp

Reply:
So, if someone calls `addCell()` with the same coordinates twice, should we output a warning? If not, this method should be called `setCell()` or `setCellValue()`.
I think that if someone needs to output a 2D table which coordinates are always the same, it can be nice to not have to reinitialize everything.